great_expectations/agent/agent.py

```diff
@@ -0,0 +1,201 @@
+            self._handle_event_as_thread_exit, event_context=event_context
+        )
+        self._current_task.add_done_callback(on_exit_callback)
+
+    def _handle_event(self, event_context: EventContext) -> EventHandlerResult:
+        """Pass events to EventHandler.
+
+        Callback passed to Subscriber.consume which forwards events to
+        the EventHandler for processing.
+
+        Args:
+            event_context: An Event with related properties and actions.
+        """
+        # warning:  this method will be executed in a different thread than
+        #           where it was defined, so take care with shared resources.
+        #           It's safe to use self._context because we restrict ourselves
+        #           to a single worker thread at any given time. The ack/nack
+        #           callback provided by the Subscriber in event_context will fail,
+        #           since it depends on the channel available in the main thread.
+
+        handler = EventHandler(context=self._context)
+        result = handler.handle_event(event_context=event_context)
+        return result
+
+    def _handle_event_as_thread_exit(
+        self, future: Future, event_context: EventContext
+    ) -> None:
+        """Callback invoked when the thread running GX exits."""
+        # this method will be invoked in the main thread after the worker exits
+
+        # get results or errors from the thread
+        error = future.exception()
+        if error is not None:
+            print("Encountered an error while running job.")
+            print(error)
+        else:
+            print("Job finished successfully.")
+            result = future.result()
+            print(result)
+
+        # TODO lakitu-139: record job as complete and send results
+
+        # ack message and cleanup resources
+        if isinstance(error, UnknownEventError):
+            # We might not have the latest Event definitions, so we requeue
+            # this event in the hope that another agent will understand it.
+            retry = True
+        else:
+            retry = False
+        event_context.event_processed(retry=retry)
+        self._current_task = None
+
+    def _can_accept_new_task(self) -> bool:
+        return self._current_task is None or self._current_task.done()
+
+    def _close_subscriber(self, subscriber: Optional[Subscriber]) -> None:
+        """Ensure the subscriber has been closed."""
+        if subscriber is None:
+            return  # nothing to close
+        try:
+            subscriber.close()
+        except SubscriberError as e:
+            print("Subscriber encountered an error while closing:")
+            print(e)
+
+    @classmethod
+    def _get_config_from_env(cls) -> GXAgentConfig:
+        """Construct GXAgentConfig from available environment variables"""
+        url = os.environ.get("BROKER_URL", None)
+        org_id = os.environ.get("GE_CLOUD_ORGANIZATION_ID", None)
```

Comment:
`GE_` has been deprecated, use `GX_`

```suggestion
        org_id = os.environ.get("GX_CLOUD_ORGANIZATION_ID", None)
```

Reply:
thanks 👍

Reply:
updated in [3e1c119](https://github.com/great-expectations/great_expectations/pull/7837/commits/3e1c119d4f38e51e18de72a6baf8e930a0aa7b0b)